ui/src/app/views/workspace-edit/component.spec.ts

```diff
@@ -87,13 +91,32 @@ describe('WorkspaceEditComponent', () => {
         // for testing.
         { provide: ActivatedRoute, useFactory: () => activatedRouteStub },
         { provide: ProfileStorageService, useValue: new ProfileStorageServiceStub() },
+        {
+          provide: CdrVersionStorageService,
+          useValue: new CdrVersionStorageServiceStub({
```

Comment:
Do we think this could be used more places than here? If so, can we make stub variables rather than use hardcoded values here? (Put the variables in the stub file as static values/exportable constants.)

Reply:
Maybe, I think we should abstract it there once there is more than one test using it. There is a price to moving the definition of a model into a utility file somewhere else, while your test is checking for some specific relationship in the data, e.g. in this test I depend on "2" being the default, and I depend on 2 mapping to name "cdr2". This would be very opaque if that were defined elsewhere.

Reply:
Perfectly happy with that. Sounds good.